### ISSUE TITLE
StrPrinter with unevaluated Mul

### ISSUE THREAD

**Issue by proy87** [sympy/sympy#21144]
`str(Add(2, Mul(-1,9,evaluate=False), evaluate=False))` outputs `2+(-1)*9` instead of `2-9`. latex printer outputs it correctly.

**Comment by oscarbenjamin** [comment]
I'm not sure what I would prefer here. Many people explicitly create unevaluated expression because they want them to display as their internal representation. The output `2+(-1)*9` explicitly shows that representation in this case.

I think it is important that unevaluated expressions do not print the same as evaluated expressions that would have a different internal form because that's just confusing. The printed result should show the difference between differently structured expressions even if they are mathematically equivalent.

**Comment by proy87** [comment]
@oscarbenjamin , makes sense.

**Comment by proy87** [comment]
I noticed that this goes from incorrect parsing of unevaluated.
parse_expr('2-9', evaluate=False) returns `Add(2, Mul(-1, 9, evaluate=False), evaluate=False)` instead of  `Add(2, -9, evaluate=False)`

**Comment by proy87** [comment]
@oscarbenjamin , `latex(Pow(3, -2, evaluate=False))` returns `\frac{1}{9}`, but should return `3^{-2}`, since it is unevaluated.
In general, latex and str don't respect unevaluated (order of arguments etc.)

**Comment by proy87** [comment]
@oscarbenjamin , there is some contradiction, because latex printer outputs `2-9`, not `2 + (-1)\cdot 9`.

**Comment by oscarbenjamin** [comment]
Yes, I think that the latex printer could be changed to be more explicit.

**Comment by proy87** [comment]
`LatexPrinter` should be improved, because it incorrectly sets brackets. `Add(3,-UnevaluatedExpr(-pi))` gives `3--\pi`. It analyzes `UnevaluatedExpr(-pi)` without minus and checks parentheses, based on coeff, not precedence.

**Comment by proy87** [comment]
Also `str(Add(3,UnevaluatedExpr(-pi)))` gives `3-pi`, not `3+(-pi)`.
